repose-aggregator/core/repose-core-api/src/main/resources/META-INF/schema/system-model/system-model.xsd

```diff
@@ -275,13 +283,170 @@
             </xs:annotation>
         </xs:attribute>
 
-        <xs:attribute name="uri-regex" type="xs:string" use="optional" default=".*">
+        <xs:attribute name="uri-regex" type="xs:string" use="optional">
             <xs:annotation>
                 <xs:documentation>
-                    <html:p>Regex of what uri is allowed access.</html:p>
+                    <html:p>
+                        Deprecated: This attribute is deprecated and will be removed in future major release.
+                        Regex of what uri is allowed access.
+                    </html:p>
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:assert vc:minVersion="1.1"
+                   xpathDefaultNamespace="##targetNamespace"
+                   test="not(@uri-regex and (count(*) > 0))"
+                   xerces:message="Cannot define both a deprecated uri-regex attribute and any of the new conditional elements"
+                   saxon:message="Cannot define both a deprecated uri-regex attribute and any of the new conditional elements"/>
+    </xs:complexType>
+
+    <xs:complexType name="FilterCriterion" abstract="true">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>Defines the parent of all the conditional filter processing elements.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+    </xs:complexType>
+
+    <xs:simpleType name="MethodList">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    A list defined as a string enumeration of HTTP method verbs as outlined in
+                    <a href="http://www.ietf.org/rfc/rfc2616.txt">RFC2616 section 9.</a>
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:list itemType="xs:string"/>
```

Comment:
Do we not want to limit the values?

Reply:
I was thinking about that. We do in one use of an HTTP method configuration and we don't in another. I don't see what limiting it does other than force us to make an update if a new verb is needed.